BitwardenShared/UI/Platform/Settings/Settings/AutoFill/PasswordAutoFill/PasswordAutoFillCoordinator.swift

```diff
@@ -0,0 +1,90 @@
+import SwiftUI
+
+// MARK: - PasswordAutoFillCoordinatorDelegate
+
+/// An object that is notified when external navigation actions need to occur from within the
+/// password autofill flow.
+///
+protocol PasswordAutoFillCoordinatorDelegate: AnyObject {
+    /// The user has completed authentication.
+    ///
+    func didCompleteAuth()
+}
+
+// MARK: - PasswordAutoFillCoordinator
+
+/// A coordinator that manages navigation for the password autofill flow.
+///
+class PasswordAutoFillCoordinator: NSObject, Coordinator, HasStackNavigator {
+    // MARK: Types
+
+    typealias Services = HasAutofillCredentialService
+        & HasConfigService
+        & HasErrorReporter
+        & HasNotificationCenterService
+        & HasStateService
+
+    // MARK: Private Properties
+
+    /// The delegate for this coordinator, used to notify when external navigation actions need to occur.
+    private weak var delegate: PasswordAutoFillCoordinatorDelegate?
+
+    /// The services used by this coordinator.
+    private let services: Services
+
+    /// The stack navigator that is managed by this coordinator.
+    private(set) weak var stackNavigator: StackNavigator?
```

Comment:
does this need to be `weak`?

Reply:
Yeah, this breaks a retain cycle. The coordinator holds a strong ref to the navigation controller, which has strong references to the view and processor. The processor has a strong ref to the coordinator, so making the coordinator's navigator weak breaks the cycle. More context [here](https://github.com/bitwarden/ios/pull/399).

Reply:
thanks for the context and PR to help explain it to me!  I appreciate that.